dot/core/service.go

```diff
@@ -260,11 +260,7 @@ func (s *Service) handleBlock(block *types.Block, state *rtstorage.TrieState) er
 
 	// store block in database
 	if err = s.blockState.AddBlock(block); err != nil {
-		if errors.Is(err, blocktree.ErrParentNotFound) && block.Header.Number != 0 {
-			return err
-		} else if errors.Is(err, blocktree.ErrBlockExists) || block.Header.Number == 0 {
-			// this is fine
-		} else {
+		if !errors.Is(err, blocktree.ErrBlockExists) && block.Header.Number != 0 {
```

Comment:
You are not checking for `ErrParentNotFound`.

Again, I am happy with the code the way it is now

Reply:
I think it is not necessary to check it because If the error is ErrParentNotFound we will return it in both versions. In both versions of the assertions, we return nil only when err is ErrBlockExists or the block checker number is 0. But I understand your suggestions and probably is better to keep the code as it is. Next time I will wait for confirmation in the issue before implementing it. Sorry for that. The PR is closed.